cmake/modules/kconfig.cmake

```diff
@@ -146,13 +147,19 @@ else()
   set(_local_TOOLCHAIN_HAS_PICOLIBC n)
 endif()
 
+# Extract git commit information for application
+git_commit_hash(${APPLICATION_SOURCE_DIR} GIT_COMMIT_HASH)
+git_commit_hash_short(${GIT_COMMIT_HASH} GIT_COMMIT_HASH_SHORT GIT_COMMIT_HASH_SHORT_INT)
```

Comment:
this then creates clashing variable names

Reply:
What is the clash? I don't see it. `GIT_COMMIT_HASH` != `GIT_COMMIT_HASH_SHORT` != `GIT_COMMIT_HASH_SHORT_INT`.

Reply:
Because this file is included in multiple places, it's a module, and you have conflicting GIT_COMMIT_HASH variables

Reply:
I'm still not seeing where the conflicting `GIT_COMMIT_HASH` variables are?
Line 151 is used to populate the hash into the variable, the value of the variable is provided on line 152 to populate the other two variable.
None of the variable names are the same...

Reply:
This file sets `GIT_COMMIT_HASH`, `GIT_COMMIT_HASH_SHORT` and `GIT_COMMIT_HASH_SHORT_INT`, those variables are then global, you can reference them anywhere after they are set, if a zephyr module also calls this function with a different folder, those same variables have been silently changed, so you indicating they are global variables with the upper case name - which they are, but there is no guarantee that when they are used, they are at the values that the user is expecting them to be at

Reply:
The only way these values could possibly change between different calls is if `APPLICATION_SOURCE_DIR` changes within a single build, in which case I suspect there would be larger issues

Reply:
I take your point about the global visibility, I will update the other PR